src/dependency-manager/src/component-config/builder.ts

```diff
@@ -69,25 +70,27 @@ export class ComponentConfigBuilder {
     return [file_path, file_contents];
   }
 
-  static async rawFromPath(path: string): Promise<{ file_path: string; file_contents: string; raw_config: RawComponentConfig }> {
-    const [file_path, file_contents] = ComponentConfigBuilder.readFromPath(path);
+  static async rawFromPath(config_path: string): Promise<{ file_path: string; file_contents: string; raw_config: RawComponentConfig }> {
+    const [file_path, file_contents] = ComponentConfigBuilder.readFromPath(config_path);
+
+    const updated_file_contents = insertFileDataFromRefs(file_contents, file_path, config_path);
 
     let raw_config;
     // Try to parse as json
     try {
-      raw_config = JSON.parse(file_contents);
+      raw_config = JSON.parse(updated_file_contents);
     } catch {
       // Try to parse as yaml
       try {
-        raw_config = yaml.safeLoad(file_contents);
+        raw_config = yaml.safeLoad(updated_file_contents);
       } catch { }
     }
 
     if (!raw_config) {
       throw new Error('Invalid file format. Must be json or yaml.');
     }
 
-    return { file_path, file_contents, raw_config };
+    return { file_path, file_contents: updated_file_contents, raw_config };
```

Comment:
Can you try updating raw_config via insertFileDataFromRefs (stringify, insert, parse)? If we keep the file_contents as is then the line numbers should be correct (def add a test). Keep the logic in the readFromPath since we wouldn't want to allow for file: reads server-side.

Reply:
I updated this from what I think you mean, but I wanted to double check that you meant `rawFromPath` instead of `readFromPath` because the line you commented on here is from `rawFromPath`